dispatch/tests/test_api_users.py

```diff
@@ -0,0 +1,222 @@
+
+        self.client.credentials()
+
+        url = reverse('api-users-list')
+
+        response = self.client.get(url, {}, format='json')
+
+        self.assertEquals(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_unauthorized_detail_get_request(self):
+        """Test that an a get request for users detail without
+        admin credentials results in a UNAUTHORIZED response"""
+
+        self.client.credentials()
+
+        # Use the id == 1, default user created on database creation
+        url = reverse('api-users-detail', args=[1])
+
+        response = self.client.get(url, {}, format='json')
+
+        self.assertEquals(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_create_user_with_password(self):
+        """Check that creating a user with a password works correctly"""
+        # NOTE: By default _create_user() supplies a good password
+        response = self._create_user("test@ubyssey.ca")
+
+        self.assertEquals(response.status_code, status.HTTP_201_CREATED)
+
+    def test_bad_passords(self):
+        """A test case to ensure a variety of bad passwords are not succesful"""
+        person_id = self._create_person("Attached Person").data['id']
+        url = reverse('api-users-list')
+        data = {
+            'email' : 'testemail123@ubyssey.ca',
+            'person_id' : person_id,
+            'password_a': 'Matching',
+            'password_b': 'NotMatching'
+        }
+        # PASSWORD TESTS
+
+        # Not matching
+        response = self.client.post(url, data, format='json')
+        self.assertEquals(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        # Too short
+        data['password_a'] = 'short'
+        data['password_b'] = 'short'
+        response = self.client.post(url, data, format='json')
+        self.assertEquals(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        # Too common
+        data['password_a'] = 'password'
+        data['password_b'] = 'password'
+        response = self.client.post(url, data, format='json')
+        self.assertEquals(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        # All numerics
+        data['password_a'] = '12345654321'
+        data['password_b'] = '12345654321'
+        response = self.client.post(url, data, format='json')
+        self.assertEquals(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_user_update(self):
+        """Ensure that user updates works correctly"""
+        response = self._create_user('test@ubyssey.ca')
+
+        UPDATED_EMAIL = 'updateTest@gmail.com'
+        UPDATED_PASSWORD = 'updatedPassword'
+
+        url = reverse('api-users-detail', args=[response.data['id']])
+        data = {
+            'email' : UPDATED_EMAIL,
+            'password_a' : UPDATED_PASSWORD,
+            'password_b' : UPDATED_PASSWORD
+        }
+        #TODO: Figure out how to test that password is updating
```

Comment:
I'm not really sure why but both passwords are failing to authenticate, might need some help on this as well